### PR TITLE
Add mono_unity_method_get_function_pointer

### PR DESCRIPTION
**Pull request by joncham**

### CODE REVIEW

File: unity/unity-embed-host/CoreCLRHost.cs [hunk 1]
```diff
@@ -168,6 +168,22 @@ public static int array_length([NativeCallbackType("MonoArray*")] IntPtr array)
...
         [NativeCallbackType("MonoClass*")] IntPtr refclass) =>
         MethodInfo.GetMethodFromHandle(method.MethodHandleFromHandleIntPtr(), RuntimeTypeHandle.FromIntPtr(refclass)).ToNativeRepresentation();
+    
+    [return: NativeCallbackType("void*")]
+    public static IntPtr unity_method_get_function_pointer([NativeCallbackType("MonoMethod*")] IntPtr method)
+    {
+        try
+        {
+
+            // TODO: Assert method is decorated with UnmanagedCallersOnly
```
Comment (mrvoorhe) [inline unity/unity-embed-host/CoreCLRHost.cs]: Should this TODO be addressed?

File: unity/unity-embed-host/CoreCLRHost.cs [hunk 2]
```diff
@@ -168,6 +168,22 @@ public static int array_length([NativeCallbackType("MonoArray*")] IntPtr array)
         [NativeCallbackType("MonoMethod*")] IntPtr method,
         [NativeCallbackType("MonoClass*")] IntPtr refclass) =>
         MethodInfo.GetMethodFromHandle(method.MethodHandleFromHandleIntPtr(), RuntimeTypeHandle.FromIntPtr(refclass)).ToNativeRepresentation();
+    
+    [return: NativeCallbackType("void*")]
+    public static IntPtr unity_method_get_function_pointer([NativeCallbackType("MonoMethod*")] IntPtr method)
+    {
+        try
```
Comment (mrvoorhe) [inline unity/unity-embed-host/CoreCLRHost.cs]: Remove try/catch.  We can handle later in a standard way.

File: unity/unity-embed-host/CoreCLRHost.cs (hunk 1 above)
Comment (joncham) [reply unity/unity-embed-host/CoreCLRHost.cs]: I'll leave it for now. Ideally we'd only check in a debug/development build and not in release.

File: unity/unity-embed-host/CoreCLRHost.cs (hunk 2 above)
Comment (joncham) [reply unity/unity-embed-host/CoreCLRHost.cs]: Done.

**Review by mrvoorhe — APPROVED** [review]

**Comment by mrvoorhe** [comment]
🟢

**Pull request merged**